src/main/java/io/appium/java_client/AppiumClientConfig.java

```diff
@@ -0,0 +1,191 @@
+     * @param directConnect If directConnect is enabled.
+     */
+    protected AppiumClientConfig(
+            URI baseUri,
+            Duration connectionTimeout,
+            Duration readTimeout,
+            Filter filters,
+            Proxy proxy,
+            Credentials credentials,
+            Boolean directConnect) {
+        super(baseUri, connectionTimeout, readTimeout, filters, proxy, credentials);
+
+        this.directConnect = Require.nonNull("Direct Connect", directConnect);
+    }
+
+    /**
+     * Return the instance of {@link AppiumClientConfig} with a default config.
+     * @return the instance of {@link AppiumClientConfig}.
+     */
+    public static AppiumClientConfig defaultConfig() {
+        return new AppiumClientConfig(
+                null,
+                DEFAULT_CONNECTION_TIMEOUT,
+                DEFAULT_READ_TIMEOUT,
+                DEFAULT_FILTER,
+                null,
+                null,
+                false);
+    }
+
+    /**
+     * Return the instance of {@link AppiumClientConfig} from the given {@link ClientConfig} parameters.
+     * @param clientConfig take a look at {@link ClientConfig}
+     * @return the instance of {@link AppiumClientConfig}.
+     */
+    public static AppiumClientConfig fromClientConfig(ClientConfig clientConfig) {
+        return new AppiumClientConfig(
+                clientConfig.baseUri(),
+                clientConfig.connectionTimeout(),
+                clientConfig.readTimeout(),
+                clientConfig.filter(),
+                clientConfig.proxy(),
+                clientConfig.credentials(),
+                false);
+    }
+
+    private AppiumClientConfig buildAppiumClientConfig(ClientConfig clientConfig, Boolean directConnect) {
+        return new AppiumClientConfig(
+                clientConfig.baseUri(),
+                clientConfig.connectionTimeout(),
+                clientConfig.readTimeout(),
+                clientConfig.filter(),
+                clientConfig.proxy(),
+                clientConfig.credentials(),
+                directConnect);
+    }
+
+    @Override
+    public AppiumClientConfig baseUri(URI baseUri) {
+        ClientConfig clientConfig = super.baseUri(baseUri);
+        return buildAppiumClientConfig(clientConfig, directConnect);
+    }
+
+    @Override
+    public AppiumClientConfig baseUrl(URL baseUrl) {
+        try {
+            return baseUri(Require.nonNull("Base URL", baseUrl).toURI());
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public AppiumClientConfig connectionTimeout(Duration timeout) {
+        ClientConfig clientConfig = super.connectionTimeout(timeout);
+        return buildAppiumClientConfig(clientConfig, directConnect);
+    }
+
+    @Override
+    public AppiumClientConfig readTimeout(Duration timeout) {
+        ClientConfig clientConfig = super.connectionTimeout(timeout);
```

Comment:
you should call super.readTimeout(timeout)

Reply:
Thank you, fixed in https://github.com/appium/java-client/pull/1959